components/atom/spinner/src/index.js

```diff
@@ -0,0 +1,106 @@
+import React, {Component} from 'react'
+import ReactDOM from 'react-dom'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import SUILoader from './SUILoader'
+
+const TYPES = {
+  FULL: 'full',
+  SECTION: 'section'
+}
+
+const DELAY = 500 // ms
+const BASE_CLASS = 'sui-atom-spinner'
+
+class AtomSpinner extends Component {
+  state = {
+    delayed: this.props.delayed
+  }
+
+  get _parentNodeClassList () {
+    if (this._parentNodeClassListCache) return this._parentNodeClassListCache
+
+    this._parentNodeClassListCache = ReactDOM.findDOMNode(this).parentNode.classList
+    return this._parentNodeClassListCache
+  }
+
+  get _parentClassName () {
+    const {type} = this.props
+
+    return cx(
+      type === TYPES.SECTION ? BASE_CLASS : `${BASE_CLASS}--full-page`
+    )
+  }
+
+  componentDidMount () {
+    if (!this.state.delayed) {
+      this.props.on && this._addParentClass()
+      return
+    }
+
+    this.timer = setTimeout(() => {
+      this.setState({delayed: false}, this._addParentClass)
+    }, DELAY)
+  }
+
+  componentWillUnmount () {
+    clearTimeout(this.timer)
+    this._removeParentClass()
+  }
+
+  _removeParentClass () {
+    this._parentNodeClassList.remove(this._parentClassName)
+  }
+
+  _addParentClass () {
+    this._parentNodeClassList.add(this._parentClassName)
+  }
+
+  componentWillReceiveProps (nextProps) {
+    if (this.props.on === nextProps.on) return
+    if (!this.props.on) return this._addParentClass()
+    if (this.props.on) return this._removeParentClass()
+  }
+
+  shouldRenderLoader () {
+    const {on} = this.props
+    const {delayed} = this.state
+
+    return !delayed && on
+  }
+
+  render () {
+    const {loader} = this.props
+    return (
+      this.shouldRenderLoader()
+        ? loader
+        : <noscript />
```

Comment:
`null` will do the trick.

Reply:
unfortunately it doesn't work with `null` as `ReactDOM.findDOMNode(this)` returns `null` and we are not able to retrieve the parent node, so we cannot set its class, so no spinner is shown